generic: enable CONFIG_ARM64_CNP and disable CONFIG_ARM64_EPAN

robimarko opened the pull request:
Enhanced Privileged Access Never (EPAN) is an ARMv8.7 feature, and since
we dont have any SoC-s implementing it lets disable it.

Ever since CONFIG_ARM64_PAN was enabled Common Not Private (CNP) is now
visible and kernel builds will stop as they are not set in kernel config
for 5.15 and 6.1.

So, lets enable Common Not Private (CNP) which is ARMv8.2 feature and will
be NOP of CPU-s that dont support it.

Fixes: https://github.com/openwrt/openwrt/commit/a2662309aae1655966d7d8f31b71ddc6edbede87 ("kernel: Enable CONFIG_ARM64_PAN to restrict kernel access to user space memory")
Signed-off-by: Robert Marko <robimarko@gmail.com>

Gingernut1978 commented:
This change is probably needed for more targets

namiltd commented:
The problem affects also more targets like imx/cortexa53, bcm4908/generic

Gingernut1978 commented:
Seems it effects the mediatek/mt7986 platform which is also ARMv8-A

robimarko commented:
Yeah, I will probably then just add it to the generic config instead.

leobsky commented:
> Yeah, I will probably then just add it to the generic config instead.

Those symbols are enabled on some targets, perhaps can be dropped too if goes to generic.
```
~/openwrt/target/linux $ grep -rn "CONFIG_ARM64_CNP"
bcm27xx/bcm2710/config-6.6:26:CONFIG_ARM64_CNP=y
bcm27xx/bcm2712/config-6.6:27:CONFIG_ARM64_CNP=y
bcm27xx/bcm2711/config-6.6:26:CONFIG_ARM64_CNP=y
armsr/armv8/config-6.6:62:CONFIG_ARM64_CNP=y
rockchip/armv8/config-6.6:25:CONFIG_ARM64_CNP=y
layerscape/armv8_64b/config-6.6:26:CONFIG_ARM64_CNP=y
layerscape/armv8_64b/config-6.1:25:CONFIG_ARM64_CNP=y

~/openwrt/target/linux $ grep -rn "CONFIG_ARM64_CNP"
bcm27xx/bcm2710/config-6.6:26:CONFIG_ARM64_CNP=y
bcm27xx/bcm2712/config-6.6:27:CONFIG_ARM64_CNP=y
bcm27xx/bcm2711/config-6.6:26:CONFIG_ARM64_CNP=y
armsr/armv8/config-6.6:62:CONFIG_ARM64_CNP=y
rockchip/armv8/config-6.6:25:CONFIG_ARM64_CNP=y
layerscape/armv8_64b/config-6.6:26:CONFIG_ARM64_CNP=y
layerscape/armv8_64b/config-6.1:25:CONFIG_ARM64_CNP=y
```

mrkiko commented:
I would turn these options off - they are not supported as far as I can tell by Cortex-A53 and Cortex-A72, why enable them? Even tough they are no-ops, isn't this a loss of memory?

hauke commented on target/linux/generic/config-6.6:
```diff
@@ -324,7 +324,9 @@ CONFIG_ARCH_MMAP_RND_COMPAT_BITS_MIN=8
 # CONFIG_ARM64_BTI is not set
+CONFIG_ARM64_CNP=y
 # CONFIG_ARM64_E0PD is not set
+CONFIG_ARM64_EPAN=y
```
This is supported starting with ARMv8.7, I think we can disable it by default.

robimarko replied:
I agree, we won't be seeing ARMv8.7 anytime soon in OpenWrt.

hauke commented on target/linux/generic/config-6.6:
```diff
@@ -324,7 +324,9 @@ CONFIG_ARCH_MMAP_RND_COMPAT_BITS_MIN=8
 # CONFIG_ARM64_64K_PAGES is not set
 # CONFIG_ARM64_AMU_EXTN is not set
 # CONFIG_ARM64_BTI is not set
+CONFIG_ARM64_CNP=y
```
This is supported since ARMv8.2 I think we should enable it even when most SoCs in OpenWrt do not support it yet, but some do.

robimarko replied:
Done

namiltd commented:
Fixed by: https://github.com/openwrt/openwrt/commit/963be1423f64156fa66a8a94cf0207ce2945bf34
but the problem still occurs for bcm4908 but it still has kernel 6.1 so just approve https://github.com/openwrt/openwrt/pull/16057

robimarko committed:
generic: 6.6: disable EPAN support

Enhanced Privileged Access Never (EPAN) is an ARMv8.7 feature, and since
we dont have any SoC-s implementing it lets disable it.

Link: https://github.com/openwrt/openwrt/pull/16211
Signed-off-by: Robert Marko <robimarko@gmail.com>

robimarko committed:
generic: 5.15, 6.1: enable CNP support

Ever since CONFIG_ARM64_PAN was enabled Common Not Private (CNP) is now
visible and kernel builds will stop as they are not set in kernel config
for 5.15 and 6.1.

So, lets enable Common Not Private (CNP) which is ARMv8.2 feature and will
be NOP of CPU-s that dont support it.

Fixes: a2662309aae1 ("kernel: Enable CONFIG_ARM64_PAN to restrict kernel access to user space memory")
Link: https://github.com/openwrt/openwrt/pull/16211
Signed-off-by: Robert Marko <robimarko@gmail.com>

robimarko commented:
Thanks! Rebased on top of main and merged!
